src/MySqlConnector/MySqlClient/ConnectionPool.cs

```diff
@@ -147,6 +164,21 @@ private ConnectionPool(ConnectionSettings cs)
 		}
 
 		static readonly ConcurrentDictionary<string, ConnectionPool> s_pools = new ConcurrentDictionary<string, ConnectionPool>();
+		static readonly TimeSpan ReapTimeSpan = TimeSpan.FromMinutes(3);
+		static readonly Task Reaper = Task.Run(async () => {
```

Comment:
In times past, I would have created a dedicated background worker thread that used `Thread.Sleep`. I haven't personally used this particular approach, but I don't see anything wrong with it.

I was wondering if we needed to use a `CancellationToken`... but there's no mechanism by which it could ever be cancelled; the connection pools live for the lifetime of the process.

(I can imagine an edge case where a consumer finishes using MySQL, calls `MySqlConnection.ClearAllPools` and would like all background code to stop running, but that seems too unlikely to be worth writing code to support.)